catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -313,8 +318,26 @@ private boolean isWhatsappInstalled() {
 		}
 	}
 
+	@Override
+	public void onReceiveResult(int resultCode, Bundle resultData) {
+		long progress;
+		if (resultCode == Constants.UPDATE_DOWNLOAD_PROGRESS) {
```

Comment:
you could use a `switch (resultCode) { case Constants.UPDATE_DOWNLOAD_PROGRESS: ...}`. and please use static imports instead of quantified ones for imports from `Constants`. (more about static imports [here](https://docs.oracle.com/javase/1.5.0/docs/guide/language/static-import.html). -> but in short just remove the `Constants` quantifier and press alt + enter select static import and let android studio do its magic).